internal/pkg/skuba/cni/cilium_test.go

```diff
@@ -0,0 +1,272 @@
+		Spec: appsv1.DaemonSetSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "fake",
+							Image: "fake",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	tests := []struct {
+		clientset   *fake.Clientset
+		daemonset   *appsv1.DaemonSet
+		errExpected bool
+		errMessage  string
+		name        string
+	}{
+		{
+			name:        "should annotate cilium daemonset with current timestamp",
+			clientset:   fake.NewSimpleClientset(),
+			daemonset:   dsBase,
+			errExpected: false,
+		},
+		{
+			name:        "should fail when daemonset does not exist",
+			clientset:   fake.NewSimpleClientset(),
+			daemonset:   dsNotExist,
+			errExpected: true,
+			errMessage:  "daemonsets.apps \"cilium\" not found",
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt // Parallel testing
+		t.Run(tt.name, func(t *testing.T) {
+			tt.clientset.AppsV1().DaemonSets(metav1.NamespaceSystem).Create(tt.daemonset)
+
+			err := AnnotateCiliumDaemonsetWithCurrentTimestamp(tt.clientset)
+
+			if tt.errExpected {
+				if err == nil {
+					t.Errorf("error expected on %s, but no error reported", tt.name)
+					return
+				}
+				if err.Error() != tt.errMessage {
+					t.Errorf("(%v) expected on %s, but different error message reported (%v)", tt.errMessage, tt.name, err.Error())
+					return
+				}
+			} else if !tt.errExpected && err != nil {
+				t.Errorf("error not expected on %s, but an error was reported (%s)", tt.name, err.Error())
+				return
+			}
+		})
+	}
+}
+
+func Test_CreateOrUpdateCiliumConfigMap(t *testing.T) {
+	cmBase := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "kubeadm-config",
+			Namespace: metav1.NamespaceSystem,
+		},
+		Data: map[string]string{"ClusterStatus": "apiEndpoints:\n  master:\n    advertiseAddress: 1.2.3.4\n    bindPort: 80\napiVersion: kubeadm.k8s.io/v1beta1\nkind: ClusterStatus\n"},
+	}
+
+	cmDefaultNamespace := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fake",
+			Namespace: metav1.NamespaceDefault,
+		},
+		Data: map[string]string{"hello": "world"},
+	}
+
+	tests := []struct {
+		clientset   *fake.Clientset
+		configmap   *corev1.ConfigMap
+		errExpected bool
+		errMessage  string
+		name        string
+		namespace   string
+	}{
+		{
+			name:        "should create or update cilium configmap",
+			clientset:   fake.NewSimpleClientset(),
+			configmap:   cmBase,
+			errExpected: false,
+		},
+		{
+			name:        "should fail when kubeadm-config configmap not exist",
+			clientset:   fake.NewSimpleClientset(),
+			configmap:   cmDefaultNamespace,
+			errExpected: true,
+			errMessage:  "unable to get api endpoints: could not retrieve the kubeadm-config configmap to get apiEndpoints: configmaps \"kubeadm-config\" not found",
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt // Parallel testing
+		t.Run(tt.name, func(t *testing.T) {
+			tt.clientset.CoreV1().ConfigMaps("kube-system").Create(tt.configmap)
```

Comment:
Is `kube-system` = `metav1.NamespaceSystem`?

Reply:
Fixed, thanks.